ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -481,22 +498,41 @@ instance ( ByronGiven
   applyChainTick (ByronEBBLedgerConfig cfg) slotNo (ByronEBBLedgerState state) =
     ByronEBBLedgerState <$> applyChainTick cfg slotNo state
 
-  applyLedgerBlock (ByronEBBLedgerConfig cfg) (ByronBlockOrEBB block)
-                   (ByronEBBLedgerState bs@(ByronLedgerState state snapshots)) =
-    case block of
-      CC.Block.ABOBBlock b -> ByronEBBLedgerState <$> applyLedgerBlock cfg (ByronBlock b) bs
-      CC.Block.ABOBBoundary b ->
-        mapExcept (fmap (\i -> ByronEBBLedgerState $ ByronLedgerState i snapshots)) $
-          return $ state
-            { CC.Block.cvsPreviousHash = Right $ CC.Block.boundaryHeaderHashAnnotated hdr
-            , CC.Block.cvsLastSlot = CC.Slot.SlotNumber $ epochSlots * CC.Block.boundaryEpoch hdr
-            }
-        where
-          hdr = CC.Block.boundaryHeader b
-          CC.Slot.EpochSlots epochSlots = given
+  applyLedgerBlock = applyByronLedgerBlockOrEBB
+    (fromBlockValidationMode CC.Block.BlockValidation)
+
+  reapplyLedgerBlock cfg blk st =
+    let validationMode = fromBlockValidationMode CC.Block.NoBlockValidation
+    in case runExcept (applyByronLedgerBlockOrEBB validationMode cfg blk st) of
+      Left  err -> error ("reapplyLedgerBlock: unexpected error: " <> show err)
+      Right st' -> st'
```

Comment:
Again I think we should comment on what's going on here, what the assumption for reapply is.